src/views/Account/Rewards/RewardsCard.tsx

```diff
@@ -0,0 +1,834 @@
+                {token.symbol} ({t('estimateAbbreviation', 'est')})
+              </span>
+            </div>
+          ) : (
+            <>
+              <TokenIcon
+                chainId={chainId}
+                address={token?.address}
+                sizeClassName='w-4 h-4'
+                className='mr-1'
+              />
+              {numberWithCommas(amount)} <span className='opacity-50 ml-1'>{token.symbol}</span>
+            </>
+          )}
+        </span>
+        <div>
+          <Tooltip id={`tooltip-promotion-${promotionId}-epoch-${awardedAt}`} tip={awardedAtLong}>
+            <span className='font-bold opacity-50'>{awardedAtShort}</span>
+          </Tooltip>
+        </div>
+      </div>
+    </li>
+  )
+}
+
+const RewardsEndInBanner = (props) => {
+  const { chainId, token, promotion } = props
+  const { t } = useTranslation()
+
+  const [days, sentence] = useRewardsEndInSentence(promotion, token)
+  const hasEnded = days <= 0
+
+  return (
+    <div
+      className='w-full flex rounded-lg my-4 font-bold shadow-sm'
+      style={{
+        backgroundImage: hasEnded
+          ? `linear-gradient(300deg, #CC3A48 0%, #AC0A48 100%)`
+          : `linear-gradient(300deg, #7617fA 0%, #CE66FF 100%)`
+      }}
+    >
+      <div
+        className={classNames(
+          'flex flex-col xs:flex-row items-center justify-center w-full text-center text-white rounded-lg p-3',
+          {
+            'bg-white bg-opacity-10': days > 0
+          }
+        )}
+      >
+        <TokenIcon
+          chainId={chainId}
+          address={token?.address}
+          sizeClassName='w-5 h-5 xs:w-4 xs:h-4'
+          className='mr-1'
+        />
+        {sentence}
+
+        {days > 1 && (
+          <Link href={{ pathname: '/deposit' }}>
+            <a className='uppercase hover:underline transition ml-2 text-pt-teal text-xs font-averta-bold'>
+              {t('depositMore', 'Deposit more')}
+            </a>
+          </Link>
+        )}
+      </div>
+    </div>
+  )
+}
+
+const useRewardsEndInSentence = (promotion, token) => {
+  const { t } = useTranslation()
+
+  const tokenSymbol = token.symbol
+  const daysRemaining = usePromotionDaysRemaining(promotion)
+
+  let rewardsEndInSentence =
+    daysRemaining < 0
+      ? t('tokenRewardsHaveEnded', '{{tokenSymbol}} have ended', { tokenSymbol })
+      : t('tokenRewardsEndSoon', '{{tokenSymbol}} rewards ending soon!', { tokenSymbol })
+
+  if (daysRemaining > 1) {
+    rewardsEndInSentence = t(
+      'tokenRewardsEndInNDays',
+      '{{tokenSymbol}} rewards end in {{days}} days',
+      {
+        tokenSymbol,
+        days: Math.round(daysRemaining)
+      }
+    )
+  }
+
+  return [daysRemaining, rewardsEndInSentence]
+}
+
+interface ClaimModalReceiptProps {
+  chainId: number
+  token: Token
+  claimableUsd: number
+  tx: Transaction
+  transactionPending: boolean
+}
+
+const ClaimModalReceipt: React.FC<ClaimModalReceiptProps> = (props) => {
```

Comment:
It'd be nice to generalize this tx pending/confirmed state a bit and update `ModalTransactionSubmitted` so it's consistent throughout the app